packages/w3up-client/src/client.js

```diff
@@ -291,9 +301,102 @@ export class Client extends Base {
         )
       }
     }
+
+    // Authorize the listed Content Serve Services to serve content from the created space
+    if (options.skipContentServeAuthorization !== true) {
+      if (
+        !options.authorizeContentServeServices ||
+        options.authorizeContentServeServices.length === 0
+      ) {
+        throw new Error(
+          'failed to authorize Content Serve Services: missing <authorizeContentServeServices> option'
+        )
+      }
+
+      if (!options.connection) {
+        throw new Error(
+          'failed to authorize Content Serve Services: missing <connection> option'
+        )
+      }
+
+      for (const service of options.authorizeContentServeServices) {
+        await this.authorizeContentServe(space, {
+          audience: service,
+          connection: options.connection,
+        })
+      }
+    }
+
     return space
   }
 
+  /**
+   * Authorizes an audience to serve content from the provided space and record egress events.
+   * It also publishes the delegation to the content serve service.
+   * Delegates the following capabilities to the audience:
+   * - `space/content/serve/*`
+   *
+   * @param {import('./types.js').OwnedSpace} space - The space to authorize the audience for.
+   * @param {object} options - Options for the authorization.
+   * @param {`did:${string}:${string}`} options.audience - The Web DID of the audience (gateway or peer) to authorize.
+   * @param {import('./types.js').ConnectionView<import('./types.js').ContentServeService>} options.connection - The connection to the Content Serve Service that will handle, validate, and store the access/delegate UCAN invocation.
+   * @param {number} [options.expiration] - The time at which the delegation expires in seconds from unix epoch.
+   */
+  async authorizeContentServe(space, options) {
```

Comment:
This is more an internal function but it would be useful to export for power users. Suggestion: move to exported function `export const authorizeContentServe = (agent, connection, space, options) => ...` and then call from `createSpace`.

Reply:
Yep, makes sense. I was thinking about implementing a feature in the Console App where users can explicitly authorize our Gateway to serve content from existing spaces, then we can use this function.

Reply:
@alanshaw, I noticed this is part of the Client class already, so a user will have access to it via `client.authorizeContentServe`. Are you thinking about moving it to an exported function in another place?

Reply:
i.e. `import { authorizeContentServe } from '@web3-storage/w3up-client'`

The client instance would use the function internally, but power users can still import and use it.

Reply:
I won't add this for now @alanshaw . We can always go back and export that if needed.